public/js/guests/controllers/guestListController.js

```diff
@@ -49,46 +44,36 @@ define([
 
         $scope.updateGuest = function (updatedGuest) {
             var alerts = validateGuest(updatedGuest),
-                id;
+                guestId;
 
             if (updatedGuest && alerts.length === 0) {
-                id = updatedGuest._id;
+                guestId = updatedGuest._id;
 
-                $http.put("/api/v1/guests/" + id, updatedGuest)
-                    .error(function (data) {
-                        $scope.alerts.push({msg: data, type: "danger"});
-                    })
-                    .success(function () {
-                        angular.forEach($scope.guests, function (guest, index) {
-                            if (guest._id === id) {
-                                $scope.guests[index] = updatedGuest;
-                            }
-                        });
+                GuestServices.update({ id:guestId }, updatedGuest, function() {
+                    angular.forEach($scope.guests, function (guest, index) {
+                        if (guest._id === guestId) {
+                            $scope.guests[index] = updatedGuest;
+                        }
                     });
+                });
```

Comment:
Hook into the error callback to add the error message. I've needed this before when the mongo DB in the cloud behaved slightly differently with id's than the local one.